Update Offline JWT Validation with Go setup to use the Okta CLI. (#644)
oktadeveloper/okta-blog: _source/_posts/2021-01-04-offline-jwt-validation-with-go.md

A file called `go.mod` containing these dependencies will be created by the `go get` command.

> To use Okta authentication you need to have a free [Okta Developer account] (https://developer.okta.com). Once you’ve done this, sign in to the developer console. 

Reviewer: You should be able to replace this line with:

```
{% include setup/cli.md type="spa" loginRedirectUri="http://localhost:8080" %}
```

> Next, create an API token. Store the value somewhere safe, but don't check it into Github! 

Reviewer: ```suggestion
Next, create an API token. Run `okta login` and open the resulting URL in your browser. Log in and go to **Security** > **API** > **Tokens**. Create a new token and store the value somewhere safe. Make sure you don't check it into GitHub! 
```

Next, create envirnonment variables for the domain and API token replacing the placeholders with your actual values from the console:

```bash
export OKTA_DOMAIN=https://dev-123456.okta.com
export OKTA_API_TOKEN=00...d
```

Next, install the [OKTA CLI](https://cli.okta.com). Just install the CLI. 

Next, login:

```bash
okta login
```

Enter your `Okta Org URL` in the form `https://dev-123456.okta.com`. Then, enter your API token.

Finally, create an Okta application:

```bash
okta apps create
```

You will be prompted for some information:
* The application name defaults to the directory name, enter: `spa-app`.
* Select Single Page App, option: 2.
* Set the default redirect URI: http://localhost:8080.
* Hit enter to select the default post logout URI: http://localhost:8080/.

A new OIDC application will be created and the configuration will be displayed. Create environment variables containing the values:

> ```bash

Reviewer: I believe you can remove lines 57-78 and the first sentence in 79. You might want to move this section up to after the `cli.md` include.

export OKTA_ISSUER_URI=https://dev-123456.okta.com/oauth2/default
export OKTA_CLIENT_ID=0...5d
```

> **PS:** The redirect and logout URIs are for local testing. In a production environment a new application needs to be created using the actual domain names.

Reviewer: ```suggestion
**PS:** The redirect and logout URIs are for local testing. In production, you'll need to update your application redirect URIs to work with your production environment.
```

Next, create an API token. Store the value somewhere safe, but don't check it into Github! 

Next, create envirnonment variables for the domain and API token replacing the placeholders with your actual values from the console:

```bash
export OKTA_DOMAIN=dev-123456.okta.com
export OKTA_API_TOKEN=00...d
```

Next, install the [OKTA CLI](https://cli.okta.com). Just install the CLI. 

Next, login:

```bash
okta login
```

Enter your `Okta Org URL` in the form `https://dev-123456.okta.com`. Then, enter your API token.

Finally, create an Okta application:

```bash
okta apps create
```

You will be prompted for some information:
* The application name defaults to the directory name, enter: `spa-app`.
* Select Single Page App, option: 2.
* Set the default redirect URI: http://localhost:8080.
* Hit enter to select the default post logout URI: http://localhost:8080/.

A new OIDC application will be created and the configuration will be displayed. Create environment variables containing the values:

```bash
export OKTA_ISSUER_URI=https://dev-123456.okta.com/oauth2/default
export OKTA_CLIENT_ID=0...5d
```

> **PS:** The redirect and logout URIs are for local testing. In a production environment a new application needs to be created using the actual domain names.

Reviewer: I believe you can remove lines 87-127.
